### DISCUSSION TITLE
Mapping keys to a sequence of commands

### DISCUSSION THREAD

**Discussion by korrat** [helix-editor/helix#11098] [Q&A] [answered]
I was trying to apply a sequence of ex mode commands from a keybinding like this:

```toml
[keys.normal.G.g]
g = [
  ":write-all",
  ":sh git gamble --pass",
  ":reload-all",
]
```

However, it seems that `:reload-all` runs before `:write-all` or `:sh …`, so my changes are undone instead of saved. Am I misusing that feature?

**Comment by kirawi** [comment, marked as answer]
IIRC, commands are run without waiting for commands like `:sh` to complete because they aren't blocking.

**Reply by korrat** [reply]
Ah, thanks, that explains it.
